### ISSUE TITLE
[Bug] MacAddress serde::from_reader deserialization fails

### ISSUE THREAD

**Issue by Deminder** [repnop/mac_address#28]
The MacAddress deserializer should also work if it can not modify the input data (for instance with serde::from_reader).  
[*The reason serde can't deserialize into a borrowed string is that it needs to modify (unescape) the encoded json, which requires ownership.*](https://github.com/serde-rs/serde/issues/1413#issuecomment-494892266)

This test will fail due to this issue:
```rust
#[test]
fn test_serde_no_borrow() {
    #[derive(Serialize, Deserialize, Debug)]
    struct Test {
        mac: MacAddress,
    }
    let json_str = serde_json::to_string(&Test {
        mac: "11:22:33:44:55:66".parse().unwrap(),
    })
    .unwrap();
    println!("{}", json_str);
    let b = json_str.as_bytes();
    let t: serde_json::Result<Test> = serde_json::from_reader(b.reader());
    if let Err(e) = t {
        panic!("serde error: {}", e);
    }
}
```
`{"mac":"11:22:33:44:55:66"}`
Error: `serde error: invalid type: string "11:22:33:44:55:66", expected a borrowed string at line 1 column 26`

If I had to guess, I would think this needs to change, right?
 https://github.com/repnop/mac_address/blob/8537b8de7fd05591c9811a3081da3fda5838d632/src/lib.rs#L69

Thanks for this nice crate @repnop

**Comment by repnop** [comment]
Thanks for the report! I'll see what I can find in the way of non-breaking changes and get this fixed sometime soon :+1:

**Comment by repnop** [comment]
I've published version `1.1.3` that uses `Cow` instead of `&str` for deserialization, shouldn't cause any breakage but allows `from_reader` to work now :)

**Comment by Deminder** [comment]
@repnop Great! Thank you
